src/test/logical_query_plan/lqp_translator_test.cpp

```diff
@@ -546,22 +515,57 @@ TEST_F(LQPTranslatorTest, ProjectionNode) {
   EXPECT_EQ(*projection_op->expressions[0], *PQPColumnExpression::from_table(*table_int_float, "a"));
 }
 
-TEST_F(LQPTranslatorTest, JoinNode) {
+TEST_F(LQPTranslatorTest, JoinNodeToJoinHash) {
+  /**
+   * Build LQP and translate to PQP
+   */
+  auto join_node = JoinNode::make(JoinMode::Inner, equals_(int_float2_b, int_float_b), int_float_node, int_float2_node);
+  const auto op = LQPTranslator{}.translate_node(join_node);
+
+  /**
+   * Check PQP
+   */
+  const auto join_op = std::dynamic_pointer_cast<JoinHash>(op);
+  ASSERT_TRUE(join_op);
+  EXPECT_EQ(join_op->primary_predicate().column_ids, ColumnIDPair(ColumnID{1}, ColumnID{1}));
+  EXPECT_EQ(join_op->primary_predicate().predicate_condition, PredicateCondition::Equals);
+  EXPECT_EQ(join_op->mode(), JoinMode::Inner);
+}
+
+TEST_F(LQPTranslatorTest, JoinNodeToJoinSortMerge) {
```

Comment:
Short note, why in this test no HashJoin is yielded?

Reply:
Or maybe on top of the three tests a short overview, that the tests get "increasingly hard" and thus fall back to "slower" implementations?

Reply:
The JoinHash cannot be used here because we have LessThan as predicate condition.

Reply:
Yes, I know, but most don’t and thus I’d like to see a short comment with the reason.
Otherwise it’s manually searching the joins and their `supports()` methods to understand this “order”.

Reply:
comment added to each test